### PR TITLE
Make reference of kubernetes component optional if referenceContent is provided

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Make reference of kubernetes component optional if referenceContent is provided.

The only tricky part in this PR that required additional attention of reviewers: we will use k8s/os component type as `IdentifiableComponentName` if alias nor reference is not provided. And as result, if Devfile contains such two components, we will show for user not the best message 
```
There are multiple components 'openshift' of type 'openshift' that cannot be ...
```
where the component type (`openshift` in this case) is duplicated.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13190

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

**Comment by garagatyi** [comment]
@benoitf I think you had an opinion on whether we should allow content without reference. Please, take a look

**Review by garagatyi — APPROVED** [review]

**Comment by sleshchenko** [comment]
ci-build

**Comment by sleshchenko** [comment]
ci-build

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java [hunk 1]
```diff
@@ -51,7 +51,12 @@ public static String getIdentifiableComponentName(Component component)
         return component.getImage();
       case KUBERNETES_COMPONENT_TYPE:
       case OPENSHIFT_COMPONENT_TYPE:
-        return component.getReference();
+        String reference = component.getReference();
+        if (reference != null) {
```
Comment (mshaposhnik) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java]: Oh. why not
`return reference != null ? reference : component.getType()`

**Review by mshaposhnik — APPROVED** [review]

**Comment by mshaposhnik** [comment]
ci-test

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java (hunk 1 above)
Comment (sleshchenko) [reply wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java]: :smile_cat:

**Comment by sleshchenko** [comment]
ci-test

**Comment by sleshchenko** [comment]
ci-build

**Pull request merged**